src/features/claims/claim-templates/hooks/use-template-claims.ts

```diff
@@ -0,0 +1,112 @@
+import { useInsecurePersistentState } from '@hedvig-ui/hooks/use-insecure-persistent-state'
+import { ClaimsFiltersType } from 'pages/claims/list/ClaimsListPage'
+import { useEffect, useState } from 'react'
+import { useHistory } from 'react-router'
+
+export interface ClaimFilterTemplate extends ClaimsFiltersType {
+  id: string
+  name: string
+}
+
+interface UseTemplateClaimsResult {
+  templateActive: boolean
+  selectedTemplate?: string
+  localFilter: ClaimsFiltersType
+  templateFilters: ClaimFilterTemplate[]
+  selectTemplate: (id: string) => void
+  createTemplate: (filter: ClaimFilterTemplate) => void
+  editTemplate: (filter: ClaimsFiltersType, id?: string) => void
+  editTemplateWithName: (filter: ClaimFilterTemplate) => void
+  removeTemplate: (id: string) => void
+}
+
+export const useTemplateClaims = (
+  selectedId?: string | null,
+): UseTemplateClaimsResult => {
+  const [templateActive, setTemplateActive] = useState(false)
+  const [selectedTemplate, setSelectedTemplate] = useState<string | undefined>()
+  const [localFilter, setLocalFilter] = useState<ClaimsFiltersType>({
+    filterClaimStates: null,
+    filterCreatedBeforeOrOnDate: null,
+    filterComplexities: null,
+    filterNumberOfMemberGroups: null,
+    filterSelectedMemberGroups: null,
+    filterMarkets: null,
+    filterTypesOfContract: null,
+  })
+
+  const [templateFilters, setTemplateFilters] = useInsecurePersistentState<
+    ClaimFilterTemplate[]
+  >('claims:templates', [])
+
+  const history = useHistory()
+
+  useEffect(() => {
+    if (!selectedId) {
+      setTemplateActive(false)
+      return
+    }
+
+    const filter = {
+      ...templateFilters.filter((template) => template.id === selectedId)[0],
+    }
+
+    if (filter) {
+      setLocalFilter(filter)
+      setTemplateActive(true)
+      setSelectedTemplate(selectedId)
+    }
+  }, [selectedId])
+
+  const selectTemplate = (id: string) => {
+    if (id === selectedTemplate) {
+      setSelectedTemplate(undefined)
+      history.push(`/claims/list/1`)
+      return
+    }
+
+    history.push(`/claims/list/1?filter=${id}`)
+  }
+
+  const createTemplate = (filter: ClaimFilterTemplate) => {
+    setTemplateFilters((prev) => [
+      ...prev,
+      {
+        ...filter,
+        name: filter.name ? filter.name : `Claims Template ${filter.id + 1}`,
+      },
+    ])
+  }
+
+  const editTemplate = (newFilter: ClaimsFiltersType, id?: string) => {
```

Comment:
Can't we do this now that we have the id in the template?

```suggestion
  const editTemplate = (newTemplate: ClaimsFiltersType) => {
```

Reply:
@rasmusguterstam
It conflicts with localFilter because this one hasn't got id. 
Local filters needs for editing in Claims List view (on editing it makes request claims with current filters, so we don't need name and id in it).
And we need id while editing templates in Dashboard